contrib/src/templates/mod.rs

```diff
@@ -140,6 +143,37 @@ pub struct TemplateInfo {
     data_type: ContentType
 }
 
+struct ManagedContext(
```

Comment:
I think I'd rather see:

```rust
#[cfg(not(debug_assertions))]
mod context {
    type Context = Context;

    impl ManagedContext {
        fn new(ctxt: Context) -> ManagedContext {
            ManagedContext(ctxt)
        }

        fn get(&self) -> impl Deref<Target=Context> {
            &self.0
        }
    }
}

#[cfg(debug_assertions)]
mod context {
    use std::sync::{RwLock, RwLockReadGuard, RwLockWriteGuard};

    struct ManagedContext(RwLock<Context>);

    impl ManagedContext {
        fn new(ctxt: Context) -> ManagedContext {
            ManagedContext(RwLock::new(ctxt))
        }

        fn get(&self) -> impl Deref<Target=Context> {
            self.0.read().unwrap()
        }

        fn get_mut(&self) -> impl DerefMut<Target=Context> {
            self.0.write().unwrap()
        }
    }
}
```

Note the use of `impl Deref`.

Reply:
I don't like that `get_mut` is only included in the `debug_assertions` case. I suppose it's fine, but it would be nice if these were symmetric.

Reply:
`get_mut` can only even exist because of the `RwLock` in `debug_assertions`; the structure is asymmetric because the release build does not need to write to `Context` and can keep the read-only form